pkg/forklift-api/webhooks/validating-webhook/admitters/plan-admitter.go

```diff
@@ -0,0 +1,121 @@
+package admitters
+
+import (
+	"context"
+	v1 "k8s.io/api/storage/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"encoding/json"
+	"fmt"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"net/http"
+
+	"github.com/konveyor/forklift-controller/pkg/apis"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1beta1"
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+)
+
+type PlanAdmitter struct {
+}
+
+func (admitter *PlanAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("Plan admitter was called")
+	raw := ar.Request.Object.Raw
+
+	plan := &api.Plan{}
+	err := json.Unmarshal(raw, plan)
+	if err != nil {
+		return util.ToAdmissionResponseError(err)
+	}
+
+	if plan.Spec.Warm {
+		log.Info("Warm migration supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Info("Couldn't get the cluster configuration", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+
+	err = api.SchemeBuilder.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't build the scheme", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+	err = apis.AddToScheme(scheme.Scheme)
+	if err != nil {
+		log.Info("Couldn't add forklift API to the scheme", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+
+	cl, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Info("Couldn't create a cluster client", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+
+	sourceProvider := api.Provider{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Source.Namespace, Name: plan.Spec.Provider.Source.Name}, &sourceProvider)
+	if err != nil {
+		log.Info("Couldn't get the source provider", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+
+	if sourceProvider.Type() == api.VSphere {
+		log.Info("Provider supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	destinationProvider := api.Provider{}
+	err = cl.Get(context.TODO(), client.ObjectKey{Namespace: plan.Spec.Provider.Destination.Namespace, Name: plan.Spec.Provider.Destination.Name}, &destinationProvider)
+	if err != nil {
+		log.Info("Couldn't get the destination provider", "error", err.Error())
+		return util.ToAdmissionResponseError(err)
+	}
+
+	if !destinationProvider.IsHost() {
+		log.Info("Remote migration supports all storages, passing")
+		return util.ToAdmissionResponseAllow()
+	}
+
+	storageClasses := v1.StorageClassList{}
+	err = cl.List(context.TODO(), &storageClasses, &client.ListOptions{})
+	if err != nil {
+		log.Info("Couldn't get the cluster storage classes", "error", err.Error())
```

Comment:
I think these can be changed to log.Error now, no?

Reply:
i based on https://github.com/kubev2v/forklift/pull/244#discussion_r1178868267

Reply:
but this is rejected now, or am I misunderstanding something?

Reply:
https://github.com/kubev2v/forklift/blob/cf2ad9146d2c17e5a88668fe73a27563e29fa2ce/pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go#L72 ?

Reply:
yeah, but in the secret admitter the request is not rejected while here it is

Reply:
true :)

Reply:
+1 on changing to `log.Error`.